### PR TITLE
2/n [TestRunner Refactor] - Enhance Browser Configuration Initialization

### PR DESCRIPTION
**Pull request by pablo-code-org**
This PR represents the second in a series of updates aimed at refactoring `runner.rb`.This one in particular removes the method to get the browsers configuration into the option parser class. 

I also added integration test that whenever someone adds/removes a config, they need to make sure the parser is able to read them correctly.

## Testing story

Added integration test that will validate the current production config

Risks:

I am moving a big folder reference, and I hope nothing breaks. Once I push the first PR 1/n, I'll run drone to validate too. I know runs doesn't run in stacked PRs

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/ui/test_runner/test_runner_options_parser.rb [hunk 1]
```diff
@@ -22,6 +23,32 @@ def parse
...
   end
 
+  def select_browser_configs
+    if @options.local
+      return [{
+        'browser' => @options.browser || 'chrome',
+                'name' => 'LocalBrowser',
+                'browserName' => @options.browser || 'chrome',
+                'version' => @options.browser_version || 'latest'
+      }]
```
Comment (Hamms) [inline dashboard/test/ui/test_runner/test_runner_options_parser.rb:33]: nit: indentation

```suggestion
      return [{
        'browser' => @options.browser || 'chrome',
        'name' => 'LocalBrowser',
        'browserName' => @options.browser || 'chrome',
        'version' => @options.browser_version || 'latest'
      }]
```

File: dashboard/test/ui/test_runner/test_runner_config_variables.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module TestRunnerConfigVariables
```
Comment (Hamms) [inline dashboard/test/ui/test_runner/test_runner_config_variables.rb:1]: I'm not sure "config variables" is accurate for these values; what about "constants"?

File: dashboard/test/ui/test_runner/test/test_runner_options_parser_test.rb [hunk 3]
```diff
@@ -1,8 +1,64 @@
 require 'minitest/autorun'
 require_relative '../test_runner_options_parser'
 class TestRunnerOptionsParserTest < Minitest::Test
+  BROWSERS_JSON_CONTENT = [
```
Comment (Hamms) [inline dashboard/test/ui/test_runner/test/test_runner_options_parser_test.rb:4]: I'm not entirely sure it's valuable to test this so exhaustively; it feels pretty silly (and potentially over-fragile) to duplicate all of our existing config here in the test like this.

I think there is opportunity here for us to write a small number of tests that check for individual, specific things we know we care about in this config. I'd definitely like to see a test like:

```ruby
  def test_mobile_vs_desktop_safari
    # Selenium uses capitalization to distinguish between mobile and desktop
    # Safari; make sure our configuration reflects that
    parser = TestRunnerOptionsParser.new(%w[-c Safari,iPhone,iPad])
    parser.parse
    assert_equal %w[Safari safari safari], desktop_parser.select_browser_configs.pluck(:browserName)
  end
```

**Review by Hamms — COMMENTED** [review]
Couple of nits, but generally I think this is heading in the right direction! I am increasingly interested in an organizational approach like [the one I suggested in the `1/n` PR](https://github.com/code-dot-org/code-dot-org/pull/55965#discussion_r1467053361), though.

File: dashboard/test/ui/test_runner/test/test_runner_options_parser_test.rb (hunk 3 above)
Comment (pablo-code-org) [reply dashboard/test/ui/test_runner/test/test_runner_options_parser_test.rb:4]: I love your approach!

I think the current test (with the long expected object)
1) Forces developers to validate the value is parsed for that specific config (Issue, this test wouldn't cover new cases, although this code doesnt change often)
2) When I wrote them my logic was "Well if someone changes, they gotta validate the code process the field properly. This is from experience where I have seen small config changes to a json or xml and somehow they break the parsing of the specific config.

Let me put some thoughts on point #1

After reading these thoughts, let me know if #2 makes sense, if not, I am not strongly married to it

**Pull request closed without merging**